### PR TITLE
Fixes #941: Add radar effect in Hints and solution

### LINKED ISSUE #941: Add radar effect in Hints and solution.
Add radar effect in hints and solution whenever a new hint appears.

Mock Links
https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/e789f7c6-c677-4c76-b3e5-d603fd84d1e1/LM-Q5-HS-Hint-1-Notification-Dot-End-

Focus on animation and hifi of this yellow radar:
Screenshot 2020-04-13 at 10 39 55 AM

### PR DESCRIPTION
**Pull request by MohamedMedhat1998**
## Explanation

- Fixes #941 
- Added the required animation
- Mocks: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/e789f7c6-c677-4c76-b3e5-d603fd84d1e1/
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/databinding/ViewBindingAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import androidx.core.animation.doOnEnd
+import androidx.databinding.BindingAdapter
+
+private val appearAnimator: ValueAnimator = ValueAnimator.ofFloat(0f, 1f)
+private val disappearAnimator: ValueAnimator = ValueAnimator.ofFloat(1f, 2f)
+private val animatorSet = AnimatorSet()
+
+@BindingAdapter("app:flashingAnimation")
+fun setFlashingAnimation(view: View, isFlashing: Boolean) {
+
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/databinding/ViewBindingAdapter.kt]: Remove this first blank line.

File: app/src/main/res/drawable/radar_moving_circle.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+  android:shape="oval">
+  <stroke
+    android:width="0.5dp"
+    android:color="#FFCD2A" />
```
Comment (rt4914) [inline app/src/main/res/drawable/radar_moving_circle.xml]: Define this color in `colors.xml`

**Review by rt4914 — CHANGES_REQUESTED** [review]
This implementation works great on tablet devices, but does not work on smaller devices like Nexus S API 25.

So, please check this animation on smaller devices.

File: app/src/main/java/org/oppia/app/databinding/ViewBindingAdapter.kt (hunk 1 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/databinding/ViewBindingAdapter.kt]: Done

File: app/src/main/res/drawable/radar_moving_circle.xml (hunk 2 above)
Comment (MohamedMedhat1998) [reply app/src/main/res/drawable/radar_moving_circle.xml]: Done

**Comment by MohamedMedhat1998** [comment]
> 
> 
> This implementation works great on tablet devices, but does not work on smaller devices like Nexus S API 25.
> 
> So, please check this animation on smaller devices.

@rt4914 It is working with me on the following devices:  
- A real device **Samsung Galaxy SM-A307FN/DS** API 29
- An emulator **Google Nexus 6** API 25 (genymotion emulator)
- An emulator **Nexus S** API 25 (android studio emulator)

**Comment by rt4914** [comment]
@anandwana001 Can you share this PR, on various devices, the main think that you have to check is the `radar` based animation on hints in explorations. You can check mock for this. 

Go to Fractions -> What is a Fraction -> Click 3 continue buttons, and in 4th screen, select `One Fifth` 2 times to get the hint at bottom-left corner. You can see the radar animation there.

**Comment by anandwana001** [comment]
> @anandwana001 Can you share this PR, on various devices, the main think that you have to check is the `radar` based animation on hints in explorations. You can check mock for this.
> 
> Go to Fractions -> What is a Fraction -> Click 3 continue buttons, and in 4th screen, select `One Fifth` 2 times to get the hint at bottom-left corner. You can see the radar animation there.

Definitely

**Comment by anandwana001** [comment]
> @anandwana001 Can you share this PR, on various devices, the main think that you have to check is the `radar` based animation on hints in explorations. You can check mock for this.
> 
> Go to Fractions -> What is a Fraction -> Click 3 continue buttons, and in 4th screen, select `One Fifth` 2 times to get the hint at bottom-left corner. You can see the radar animation there.

@MohamedMedhat1998 @rt4914 I tried this PR with various devices:
1. Nexus S 480*800: hdpi - worked perfectly
2. Pixel 1080*1920: 420dpi- worked perfectly
3. Pixel XL 1080*1920: xxhdpi -  worked perfectly
4. 7 WSVGA Tablet 768*1024hdpi - worked perfectly
5. Redmi Note 5 Pro 1080 x 2160 pixels- worked Perfectly

**Review by rt4914 — APPROVED** [review]
LGTM

**Pull request merged**